runtime/process.go

```diff
@@ -262,10 +262,27 @@ func (p *process) handleSigkilledShim(rst uint32, rerr error) (uint32, error) {
 		}
 		if ppid == "1" {
 			logrus.Warnf("containerd: %s:%s shim died, killing associated process", p.container.id, p.id)
+			// Before sending SIGKILL to container, we need to make sure
+			// the container is not in Paused state. If the container is
+			// Paused, the container will not response to any signal
+			// we should Resume it after sending SIGKILL
+			var (
+				s    State
+				err1 error
+			)
+			if p.container != nil {
+				s, err1 = p.container.Status()
```

Comment:
It's unlikely, but if you were to resume here and `SIGKILL` fails somehow, you end up with a running process.

Also, if resuming the container first and this thread somehow gets unscheduled by the kernel, then the processes in the container have the time to do stuff which may not be desirable.